### PR TITLE
Replace usage of substr with str_starts_with and str_ends_with

### PR DESCRIPTION
**Pull request by Soean**
Trac ticket: https://core.trac.wordpress.org/ticket/58220

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

**Comment by spacedmonkey** [comment]
@Soean There seems to be a unit test failure for php 8.1

**Comment by Soean** [comment]
@spacedmonkey I merged trunk, now the tests look good.

**Review by spacedmonkey — APPROVED** [review]
Looks good to me!

**Comment by joelataccelm** [comment]
wait, how are you doing this and also still claiming to support php 7.4? these aren't supported in php 7.4 and will fatal error.

**Comment by spacedmonkey** [comment]
@joelataccelm  A backward compatibility shim was added https://github.com/WordPress/wordpress-develop/commit/3cc8f1237a737ccf895682029afacafd7305d420 ( in WP 5.9.0 ). These are already in core in many many places. WordPress currently supports down to PHP 5.6.

**Comment by joelataccelm** [comment]
> @joelataccelm A backward compatibility shim was added [3cc8f12](https://github.com/WordPress/wordpress-develop/commit/3cc8f1237a737ccf895682029afacafd7305d420) ( in WP 5.9.0 ). These are already in core in many many places. WordPress currently supports down to PHP 5.6.

got it, sorry. I just spotted that in the latest compat file. the version I noticed this fatal erroring on doesn't have the str_ functions in the compat file, so I just need to upgrade.

**Comment by sabernhardt** [comment]
According to the [Hello Dolly plugin page](https://wordpress.org/plugins/hello-dolly/), it should be compatible with WordPress 4.6 or higher. The `str_starts_with` function would require either PHP8 or WP5.9.

File: src/wp-content/plugins/hello.php [hunk 1]
```diff
@@ -53,7 +53,7 @@ function hello_dolly_get_lyric() {
 function hello_dolly() {
 	$chosen = hello_dolly_get_lyric();
 	$lang   = '';
-	if ( 'en_' !== substr( get_user_locale(), 0, 3 ) ) {
+	if ( ! str_starts_with( get_user_locale(), 'en_' ) ) {
```
Comment (spacedmonkey) [inline src/wp-content/plugins/hello.php]:
```suggestion
	if ( 'en_' !== substr( get_user_locale(), 0, 3 ) ) {
```

**Review by spacedmonkey — APPROVED** [review]

**Comment by SergeyBiryukov** [comment]
Thanks for the PR! Merged in [r55990](https://core.trac.wordpress.org/changeset/55990).

**Pull request closed without merging**